### PR TITLE
Optimization & bug fixes for PerfGraph

### PR DESCRIPTION
**Pull request by pushfoo**
**tl;dr ~7% faster, fixes bugs, backward compatible**

These changes seem to run at least 7% faster and with far more FPS stability than the current development branch. Most of the improvement is likely from the removal of `arcade.draw_text`.

Major changes:
* Switch to `arcade.Text` instances instead of `arcade.draw_text` calls
* Use batched mutable `pyglet.graphics.Line` instances for non-data lines
* Convert multiple attributes into backward-compatible properties that update mutable pyglet shapes
* Fix rendering bugs when PerfGraph instances are scaled

## Performance comparison:
These screen shots are toward the maximum improvement (52 FPS is the minimum on development for my system). 7% improvement seems to be the average. Getting exact numbers doesn't seem worth the time at the moment, but the improved FPS stability of the PR version is worth noting.

Before:
image
After:
image

### CODE REVIEW

**Comment by einarf** [comment]
Let's just merge this. It will be be first feature using pyglet batches in arcade 👍 (Since this is now possible in 2.7)
